experiments/ClimaEarth/components/ocean/slab_ocean.jl

```diff
@@ -203,3 +218,78 @@ Perform DSS on the state of a component simulation, intended to be used
 before the initial step of a run. This method acts on slab ocean model sims.
 """
 dss_state!(sim::SlabOceanSimulation) = CC.Spaces.weighted_dss!(sim.integrator.u, sim.integrator.p.dss_buffer)
+
+"""
+    FieldExchanger.update_sim!(::SlabOceanSimulation, csf, area_fraction)
+
+Updates the air density (needed for the turbulent flux calculation)
+and the direct and diffuse albedos of the ocean.
+"""
+function FieldExchanger.update_sim!(sim::SlabOceanSimulation, csf, area_fraction)
+    update_field!(sim, Val(:air_density), csf.ρ_sfc)
+    update_field!(sim, Val(:u_atmos), csf.u_int)
+    update_field!(sim, Val(:v_atmos), csf.v_int)
+
+    # Update the direct and diffuse albedos with the new atmospheric wind
+    set_albedos!(sim, sim.integrator.t)
+end
+
+"""
+    FieldExchanger.import_atmos_fields!(csf, sim::SlabOceanSimulation, atmos_sim)
+
+Imports quantities from the coupled simulation fields into the ocean simulation.
+This is similar to the default method defined in FieldExchanger.jl, but it also
+includes atmospheric wind, which is required to compute the ocean albedo.
+"""
+function FieldExchanger.import_atmos_fields!(csf, sim::SlabOceanSimulation, atmos_sim)
+    # surface density - needed for q_sat and requires atmos and sfc states, so it is calculated and saved in the coupler
+    Interfacer.remap!(csf.ρ_sfc, FluxCalculator.calculate_surface_air_density(atmos_sim, csf.T_sfc)) # TODO: generalize for PartitionedStateFluxes (#445) (use individual T_sfc)
+
+    # radiative fluxes
+    Interfacer.get_field(csf.F_radiative, atmos_sim, Val(:radiative_energy_flux_sfc))
+
+    # precipitation
+    Interfacer.get_field(csf.P_liq, atmos_sim, Val(:liquid_precipitation))
+    Interfacer.get_field(csf.P_snow, atmos_sim, Val(:snow_precipitation))
+
+    # wind
+    Interfacer.get_field(csf.u_int, atmos_sim, Val(:u_int))
+    Interfacer.get_field(csf.v_int, atmos_sim, Val(:v_int))
+end
+
+"""
+    function set_albedos!(sim::SlabOceanSimulation, t)
+
+Set the direct and diffuse albedos of the ocean based on the current date and
+the atmospheric wind. The albedos are calculated using the `surface_albedo_direct`
+and `surface_albedo_diffuse` functions from the `ClimaAtmos` module, so this
+is dependent on running with `ClimaAtmosSimulation` as the atmosphere simulation.
+"""
+function set_albedos!(sim::SlabOceanSimulation, t)
+    u = sim.integrator.u
+    p = sim.integrator.p
+    FT = eltype(u)
```

Comment:
```suggestion
    FT = CC.Spaces.undertype(axes(u))
```
This ensures that we get the `FT` even when `u` is not a scalr